assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -7264,6 +7461,12 @@ else if (sameAssignmentTitleInContext(assignmentRef, title, (String) state.getAt
 		else b = Boolean.TRUE.toString();
 		state.setAttribute(NEW_ASSIGNMENT_ALLOW_STUDENT_VIEW, b);
 		
+		//set whether students can view the review service grades
+		r = params.getString(NEW_ASSIGNMENT_ALLOW_STUDENT_VIEW_EXTERNAL_GRADE);
+		if (r == null) b = Boolean.FALSE.toString();
+		else b = Boolean.TRUE.toString();
```

Comment:
This class is huge, so we should take every opportunity to condense things here. I'd change these if.. else checks to one liners like:
b = String.valueOf(r != null);